lib/local-storage.js

```diff
@@ -484,158 +499,122 @@ Storage.prototype.get_package = function(name, options, callback) {
+//  storages[self.config.storage] = true
 //
-// Arguments:
-// - name - package name
-// - updateFn - function(package, cb) - update function
-// - callback - callback that gets invoked after it's all updated
+//  if (self.config.packages) {
+//    Object.keys(self.packages || {}).map(function (pkg) {
+//      if (self.config.packages[pkg].storage) {
+//        storages[self.config.packages[pkg].storage] = true
+//      }
+//    })
+//  }
 //
-// Algorithm:
-// 1. lock package.json for writing
-// 2. read package.json
-// 3. updateFn(pkg, cb), and wait for cb
-// 4. write package.json.tmp
-// 5. move package.json.tmp package.json
-// 6. callback(err?)
+//  var base = Path.dirname(self.config.self_path);
 //
-Storage.prototype.update_package = function(name, updateFn, _callback) {
-  var self = this
-  var storage = self.storage(name)
-  if (!storage) return _callback( Error[404]('no such package available') )
-  storage.lock_and_read_json(info_file, function(err, fd, json) {
-    function callback() {
-      var _args = arguments
-      if (fd) {
-        fs.close(fd, function(err) {
-          if (err) return _callback(err)
-          _callback.apply(null, _args)
-        })
-      } else {
-        _callback.apply(null, _args)
-      }
-    }
+//  async.eachSeries(Object.keys(storages), function (storage, cb) {
+//    fs.readdir(Path.resolve(base, storage), function (err, files) {
+//      if (err) return cb(err)
+//
+//      async.eachSeries(files, function (file, cb) {
+//        if (file.match(/^@/)) {
+//          // scoped
+//          fs.readdir(Path.resolve(base, storage, file), function (err, files) {
+//            if (err) return cb(err)
+//
+//            async.eachSeries(files, function (file2, cb) {
+//              if (Utils.validate_name(file2)) {
+//                on_package({
+//                  name: file + '/' + file2,
+//                  path: Path.resolve(base, storage, file, file2),
+//                }, cb)
+//              } else {
+//                cb()
+//              }
+//            }, cb)
+//          })
+//        } else if (Utils.validate_name(file)) {
+//          on_package({
+//            name: file,
+//            path: Path.resolve(base, storage, file)
+//          }, cb)
+//        } else {
+//          cb()
+//        }
+//      }, cb)
+//    })
+//  }, on_end)
+//}
 
-    if (err) {
-      if (err.code === 'EAGAIN') {
-        return callback( Error[503]('resource temporarily unavailable') )
-      } else if (err.code === 'ENOENT') {
-        return callback( Error[404]('no such package available') )
-      } else {
-        return callback(err)
-      }
-    }
+//
 
-    self._normalize_package(json)
-    updateFn(json, function(err) {
-      if (err) return callback(err)
+Storage.prototype.update_package = function(package, updateFn, _callback) {
+  // TODO -- is this function missing. I'm not certain how the previous code we wouldn't have storage.
```

Comment:
`storage` can be specified in the config as `null` which means "don't store this package anywhere". A few people asked for this feature.